velox/connectors/hive/TableHandle.h

```diff
@@ -30,15 +30,41 @@ class HiveColumnHandle : public ColumnHandle {
  public:
   enum class ColumnType { kPartitionKey, kRegular, kSynthesized };
 
+  /// NOTE: 'dataType' is the column type in target write table. 'hiveType' is
+  /// converted type of the corresponding column in source table which might not
+  /// be the same type, and the table scan needs to do data coercion if needs.
+  /// The table writer also needs to respect the type difference when processing
+  /// input data such as bucket id calculation.
   HiveColumnHandle(
       const std::string& name,
       ColumnType columnType,
       TypePtr dataType,
+      TypePtr hiveType,
       std::vector<common::Subfield> requiredSubfields = {})
       : name_(name),
         columnType_(columnType),
         dataType_(std::move(dataType)),
-        requiredSubfields_(std::move(requiredSubfields)) {}
+        hiveType_(std::move(hiveType)),
+        requiredSubfields_(std::move(requiredSubfields)) {
+    VELOX_CHECK(
```

Comment:
Shouldn't this be a USER check since the problem is that we don't support different types for now?